### PR TITLE
Add semantic reranking notebook

### PR DESCRIPTION
**Pull request by demjened**
Add notebook on semantic reranking (BM25 search + reranking with Cohere through inference endpoint).

### CODE REVIEW

File: notebooks/search/10-semantic-reranking.ipynb [hunk 1]
```diff
@@ -0,0 +1,576 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "24f0133923553d28",
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "url = \"https://raw.githubusercontent.com/elastic/elasticsearch-labs/main/notebooks/search/articles-wikipedia.json\"\n",
```
Comment (demjened) [inline notebooks/search/10-semantic-reranking.ipynb:328]: If testing before this PR is merged, use this URL:
https://raw.githubusercontent.com/elastic/elasticsearch-labs/2b16684bce3abb31f0e6b20c9a52b0f3208baa2c/notebooks/search/articles-wikipedia.json

File: notebooks/search/10-semantic-reranking.ipynb [hunk 2]
```diff
@@ -0,0 +1,576 @@
...
+    "try:\n",
+    "    client.options(\n",
+    "        request_timeout=60, max_retries=3, retry_on_timeout=True\n",
+    "    ).inference.put_model(\n",
+    "        task_type=\"rerank\",\n",
+    "        inference_id=\"cohere-rerank-inference\",\n",
+    "        model_config={\n",
+    "            \"service\": \"cohere\",\n",
+    "            \"service_settings\": {\n",
+    "                \"api_key\": \"kiOfsXJV2kMQ46xzve6snydXhKtfnfLI8pCgCdm0\",\n",
```
Comment (joshdevins) [inline notebooks/search/10-semantic-reranking.ipynb:261]: Change to a placeholder value, e.g. `PLACEHOLDER` or `API_KEY`

**Review by joshdevins — APPROVED** [review]
Minor change only, otherwise LGTM

**Comment by demjened** [comment]
@leemthompo With regards to mentioning the other Cohere notebook, how about this text?

_This example will show how to combine search and [semantic reranking](https://www.elastic.co/guide/en/elasticsearch/reference/current/semantic-reranking.html) to improve the accuracy of your search results. We'll be using the [rerank feature from Cohere](https://cohere.com/rerank)._

_Note: for a complete integration with Cohere please refer to [this notebook](https://github.com/elastic/elasticsearch-labs/blob/main/notebooks/integrations/cohere/cohere-elasticsearch.ipynb). This example focuses on Cohere reranking only through an Elastic [retriever](https://www.elastic.co/guide/en/elasticsearch/reference/current/retrievers-overview.html) query._

**Comment by leemthompo** [comment]
Sounds good. Maybe rename the notebook to semantic-reranking-retrievers-cohere.ipynb because we'll be adding at least one more semantic reranking with retrievers notebook using different providers/Eland option.

**Pull request merged**